subsystem/kubeapi_test.go

```diff
@@ -647,15 +648,68 @@ var _ = Describe("[kube-api]cluster installation", func() {
 
 		infraEnv := getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName)
 		firstURL := infraEnv.Status.ISODownloadURL
+		firstCreatedAt := infraEnv.Status.CreatedAt
+
+		annotations := map[string]string{"foo": "bar"}
+		By("Update InfraEnv Annotations")
+		Eventually(func() error {
+			infraEnv.SetAnnotations(annotations)
+			return kubeClient.Update(ctx, infraEnv)
+		}, "30s", "10s").Should(BeNil())
+
+		Consistently(func() map[string]string {
+			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).GetAnnotations()
+		}, "30s", "2s").Should(Equal(annotations))
+
+		By("Verify InfraEnv Status ISODownloadURL has not changed")
+		Consistently(func() string {
+			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).Status.ISODownloadURL
+		}, "30s", "2s").Should(Equal(firstURL))
+
+		By("Verify InfraEnv Status CreatedAt has not changed")
+		Consistently(func() *metav1.Time {
+			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).Status.CreatedAt
+		}, "30s", "2s").Should(Equal(firstCreatedAt))
+	})
+
+	It("verify InfraEnv ISODownloadURL and image CreatedAt are changing  - update IgnitionConfigOverride", func() {
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		deployAgentClusterInstallCRD(ctx, kubeClient, aciSpec, clusterDeploymentSpec.ClusterInstallRef.Name)
+		By("Deploy InfraEnv")
+		deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+
+		infraEnvKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      infraNsName.Name,
+		}
 
+		Eventually(func() string {
+			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).Status.ISODownloadURL
+		}, "15s", "5s").Should(Not(BeEmpty()))
+
+		infraEnv := getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName)
+		firstURL := infraEnv.Status.ISODownloadURL
+		firstCreatedAt := infraEnv.Status.CreatedAt
+		By("Update InfraEnv IgnitionConfigOverride")
 		Eventually(func() error {
-			infraEnv.SetAnnotations(map[string]string{"foo": "bar"})
+			infraEnvSpec.IgnitionConfigOverride = fakeIgnitionConfigOverride
+			infraEnv.Spec = *infraEnvSpec
 			return kubeClient.Update(ctx, infraEnv)
 		}, "30s", "10s").Should(BeNil())
 
+		Consistently(func() string {
+			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).Spec.IgnitionConfigOverride
+		}, "30s", "2s").Should(Equal(fakeIgnitionConfigOverride))
+
+		By("Verify InfraEnv Status ISODownloadURL has changed")
 		Consistently(func() string {
 			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).Status.ISODownloadURL
-		}, "10s", "2s").Should(Equal(firstURL))
+		}, "30s", "2s").ShouldNot(Equal(firstURL))
+
+		By("Verify InfraEnv Status CreatedAt has changed")
+		Consistently(func() *metav1.Time {
```

Comment:
can be replaced with expect beacuse eventually that will validate the URL will make sure that everything already need to be set.